File: mobile/lib/models/mission.dart
// lib/models/mission.dart
// Représente une mission (course) assignée au coursier.

class Mission {
  final int id;
  final String adresseDepart;
  final double latitudeDepart;
  final double longitudeDepart;
  final String adresseArrivee;
  final double latitudeArrivee;
  final double longitudeArrivee;
  final String? description;
  String statut;
  final String priorite;

  Mission({
    required this.id,
    required this.adresseDepart,
    required this.latitudeDepart,
    required this.longitudeDepart,
    required this.adresseArrivee,
    required this.latitudeArrivee,
    required this.longitudeArrivee,
    this.description,
    required this.statut,
    required this.priorite,
  });

  factory Mission.fromJson(Map<String, dynamic> json) {
    return Mission(
      id: json['id'],
      adresseDepart: json['adresse_depart'] ?? '',
      latitudeDepart: double.tryParse(json['latitude_depart'].toString()) ?? 0,
      longitudeDepart: double.tryParse(json['longitude_depart'].toString()) ?? 0,
      adresseArrivee: json['adresse_arrivee'] ?? '',
      latitudeArrivee: double.tryParse(json['latitude_arrivee'].toString()) ?? 0,
      longitudeArrivee: double.tryParse(json['longitude_arrivee'].toString()) ?? 0,
      description: json['description'],
      statut: json['statut'] ?? 'en_attente',
      priorite: json['priorite'] ?? 'normale',
    );
  }
}

File: backend/src/models/Mission.js
// src/models/Mission.js
// Une mission (course) à assigner à un coursier disponible.

module.exports = (sequelize, DataTypes) => {
  const Mission = sequelize.define('Mission', {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true,
    },
    coursier_id: {
      type: DataTypes.INTEGER,
      allowNull: true, // null tant que non assignée
    },
    adresse_depart: {
      type: DataTypes.STRING(255),
      allowNull: false,
    },
    latitude_depart: {
      type: DataTypes.DECIMAL(10, 7),
      allowNull: false,
    },
    longitude_depart: {
      type: DataTypes.DECIMAL(10, 7),
      allowNull: false,
    },
    adresse_arrivee: {
      type: DataTypes.STRING(255),
      allowNull: false,
    },
    latitude_arrivee: {
      type: DataTypes.DECIMAL(10, 7),
      allowNull: false,
    },
    longitude_arrivee: {
      type: DataTypes.DECIMAL(10, 7),
      allowNull: false,
    },
    description: {
      type: DataTypes.TEXT,
      allowNull: true,
    },
    statut: {
      type: DataTypes.ENUM('en_attente', 'assignee', 'en_cours', 'terminee', 'annulee'),
      allowNull: false,
      defaultValue: 'en_attente',
    },
    priorite: {
      type: DataTypes.ENUM('normale', 'urgente'),
      defaultValue: 'normale',
    },
    assignee_at: {
      type: DataTypes.DATE,
      allowNull: true,
    },
    terminee_at: {
      type: DataTypes.DATE,
      allowNull: true,
    },
  }, {
    tableName: 'missions',
    indexes: [
      { fields: ['statut'] },
      { fields: ['coursier_id'] },
    ],
  });

  return Mission;
};
